drivers/st25r3911b/st25r3911b.c

```diff
@@ -0,0 +1,570 @@
+				    reg_data[1]);
+}
+
+int st25r3911b_mask_receive_timer_set(u32_t fc)
+{
+	LOG_DBG("Set mask receive timer to %u fc", fc);
+	return st25r3911b_reg_write(ST25R3911B_REG_MASK_RX_TIM,
+				    ST25R3911B_FC_TO_64FC(fc));
+}
+
+int st25r3911b_init(void)
+{
+	int err;
+
+	err = st25r3911b_spi_init();
+	if (err) {
+		return err;
+	}
+
+	/* Set default settings */
+	err = st25r3911b_cmd_execute(ST25R3911B_CMD_SET_DEFAULT);
+	if (err) {
+		return err;
+	}
+
+	/* By default all interrupts are on, so disable them */
+	err = st25r3911b_irq_disable(ST25R3911B_IRQ_MASK_ALL);
+	if (err) {
+		return err;
+	}
+
+	/* Clear all interrupts */
+	err = st25r39_irq_clear();
+	if (err) {
+		return err;
+	}
+
+	/* Start oscillator and wait for it to be stable */
+	err = osc_start();
+	if (err) {
+		LOG_ERR("Oscillator start failed");
+		return err;
+	}
+
+	u32_t mV = 0;
+
+	/* Measure a supply voltage to detect if device
+	 * is powered by 5V or 3.3V.
+	 */
+	err = measure_voltage(ST25R3911B_REG_REGULATOR_CTRL_MPSV_VDD, &mV);
+	if (err) {
+		LOG_ERR("Voltage measure failed");
+		return err;
+	}
+
+	/* Set power supply 5V or 3V3 */
+	err = st25r3911b_reg_modify(ST25R3911B_REG_IO_CONF2,
+				    ST25R3911B_REG_IO_CONF2_SUP3V,
+				    (mV < POWER_SUPP_3V3_MAX_LVL) ?
+				    ST25R3911B_REG_IO_CONF2_SUP3V : 0);
+	if (err) {
+		LOG_ERR("Power supply voltage level set failed");
+		return err;
+	}
+
+	/* Adjust voltage regulator. */
+	err = adjust_regulator(&mV);
+	if (err) {
+		LOG_ERR("Regulator adjust failed");
+		return err;
+	}
+
+	/* Calibrate antenna, according to errata
+	 * this should be done always twice
+	 */
+	err = calibrate_antenna();
+	if (err) {
+		LOG_ERR("Antenna calibration failed");
+		return err;
+	}
+
+	err = calibrate_antenna();
+	if (err) {
+		LOG_ERR("Antenna calibration failed");
+		return err;
+	}
+
+	/* Adjust regulator after antenna calibration */
+	err = adjust_regulator(&mV);
+	if (err) {
+		LOG_ERR("Regulator adjust failed");
+		return err;
+	}
+
+	/* Disable RX and TX mode */
+	err = st25r3911b_rx_tx_disable();
+	if (err) {
+		return err;
+	}
+
+	/* Clear FIFO */
+	return st25r3911b_cmd_execute(ST25R3911B_CMD_CLEAR);
+}
+
+int st25r3911b_tx_len_set(u16_t len)
+{
+	int err;
+	u8_t msb_val;
+
+	if (len > ST25R3911B_MAX_TX_LEN) {
+		return -EFAULT;
+	}
+
+	msb_val = (len >> ST25R3911B_REG_NUM_TX_BYTES_NTX_SHIFT_LSB) & 0xFF;
+
+	err = st25r3911b_reg_modify(ST25R3911B_REG_NUM_TX_BYTES_REG2,
+				    ST25R3911B_REG_NUM_TX_BYTES_REG2_NTX_MASK,
+				    (len << ST25R3911B_REG_NUM_TX_BYTES_NTX_SHIFT) &
+				    ST25R3911B_REG_NUM_TX_BYTES_REG2_NTX_MASK);
+	if (err) {
+		return err;
+	}
+
+	err = st25r3911b_reg_write(ST25R3911B_REG_NUM_TX_BYTES_REG1,
+				   msb_val);
+	if (!err) {
+		LOG_DBG("Fifo Tx length set to %u", len);
+	}
+
+	return err;
+}
+
+int st25r3911b_field_on(u8_t collision_threshold, u8_t peer_threshold,
+			u8_t delay)
+{
+	int err;
+	u32_t status;
+
+	if (collision_threshold != ST25R3911B_NO_THRESHOLD_ANTICOLLISION) {
```

Comment:
why `ST25R3911B_NO_THRESHOLD_ANTICOLLISION` is not handler? how this is disabled?

Reply:
I added write the default registers value in this case.